streams/src/test/java/org/apache/kafka/streams/integration/StoresIntegrationTest.java

```diff
@@ -0,0 +1,128 @@
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.streams.integration;
+import org.apache.kafka.clients.consumer.ConsumerConfig;
+import org.apache.kafka.common.serialization.Serdes;
+import org.apache.kafka.streams.KafkaStreams;
+import org.apache.kafka.streams.StreamsConfig;
+import org.apache.kafka.streams.processor.Processor;
+import org.apache.kafka.streams.processor.ProcessorContext;
+import org.apache.kafka.streams.processor.ProcessorSupplier;
+import org.apache.kafka.streams.processor.TopologyBuilder;
+import org.apache.kafka.streams.state.Stores;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.ClassRule;
+import static org.junit.Assert.assertEquals;
+import java.util.Properties;
+
+import org.apache.kafka.streams.integration.utils.EmbeddedSingleNodeKafkaCluster;
+
+
+public class StoresIntegrationTest {
+    @ClassRule
+    public static EmbeddedSingleNodeKafkaCluster cluster = new EmbeddedSingleNodeKafkaCluster();
+    private static final String INPUT_TOPIC = "inputTopic";
+    private static final String OUTPUT_TOPIC = "outputTopic";
+    private static int foundStores = 0;
+
+    @BeforeClass
+    public static void startKafkaCluster() throws Exception {
+        cluster.createTopic(INPUT_TOPIC);
+        cluster.createTopic(OUTPUT_TOPIC);
+    }
+
+    private static class MyProcessorSupplier implements ProcessorSupplier<String, String> {
+        private final int numStores;
+        MyProcessorSupplier(int numStores) {
+            this.numStores = numStores;
+        }
+        @Override
+        public Processor<String, String> get() {
+            return new Processor<String, String>() {
+                private ProcessorContext context;
+
+                @Override
+                @SuppressWarnings("unchecked")
+                public void init(ProcessorContext context) {
+                    this.context = context;
+                    this.context.schedule(1000);
+                    for (int i = 0; i < numStores; i++) {
+                        String storeName = "Counts" + i;
+                        if (context.getStateStore(storeName) != null) {
+                            foundStores++;
+                        }
+                    }
+                }
+
+                @Override
+                public void process(String dummy, String line) {
+                    // do nothing
```

Comment:
Nitpick: Hmm, why the "do nothing" comments?  One can clearly see that there's no code.  (I could understand the necessity of a comment if the comment were to justify *why* there's nothing being done here.)